play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/snippets/GovukFormGroupSnippet.scala.html

```diff
@@ -0,0 +1,25 @@
+@*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import uk.gov.hmrc.govukfrontend.views.viewmodels.FormGroup
+
+@this()
+
+@(formGroup: FormGroup = FormGroup(), errorMessage: Option[ErrorMessage] = None)(content: => Html)
```

Comment:
feels a bit off passing the error message - what about a boolean and pass errorMessage.isDefined?

or considered if we could just update the FormGroup(classes=) but not sure that feels right either - nor does popping in an error flag to FormGroup case class

Reply:
I kind of wonder if we should just put it in components as GovukFormGroup.scala.html too - I know technically it's not a component from govuk but 🤷‍♂️

Reply:
> feels a bit off passing the error message - what about a boolean and pass errorMessage.isDefined?
> 
> or considered if we could just update the FormGroup(classes=) but not sure that feels right either - nor does popping in an error flag to FormGroup case class

Just have a `hasErrorMessage` boolean then?  Or could even check the `content` in the snippet to see if it contains `govuk-error-message`...? 🤔  Though that may be too brittle 🤷

Reply:
> I kind of wonder if we should just put it in components as GovukFormGroup.scala.html too - I know technically it's not a component from govuk but 🤷‍♂️

I wanted to make it clear that it's not a first-class component, but I'm not really precious about the current naming.

Reply:
yeah, maybe hasErrorMessage 🤷‍♂️ the more I read it the more I don't mind

we do have some other "internal" type components in the main components folder (the ones to control content layout), but then I don't know if they should really be there

I guess having snippets folder we could have a README to say "these are internal, not a public api, use at own risk as we might rejig as needed"

Reply:
Another suggestion: rather than `snippets` as folder / package, how about `helpers`? Then we're not coupling the package too tightly for future useful things, plus matches the `hmrcfrontend` naming convention.